kernel-spark/src/main/java/io/delta/kernel/spark/utils/RowTrackingUtils.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.spark.utils;
+
+import io.delta.kernel.internal.TableConfig;
+import io.delta.kernel.internal.actions.Metadata;
+import io.delta.kernel.internal.actions.Protocol;
+import io.delta.kernel.internal.tablefeatures.TableFeatures;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.spark.sql.catalyst.expressions.FileSourceConstantMetadataStructField;
+import org.apache.spark.sql.catalyst.expressions.FileSourceGeneratedMetadataStructField;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.MetadataBuilder;
+import org.apache.spark.sql.types.StructField;
+
+/**
+ * Utility methods for row tracking in Kernel based connector. This class provides row tracking
+ * functionality with Spark-specific metadata attributes for marking metadata columns as constant or
+ * generated fields.
+ */
+public class RowTrackingUtils {
+
+  // Field names for row tracking metadata columns (matching Spark V1 definitions)
+  private static final String BASE_ROW_ID = "base_row_id";
+  private static final String ROW_ID = "row_id";
+  private static final String DEFAULT_ROW_COMMIT_VERSION = "default_row_commit_version";
+  private static final String ROW_COMMIT_VERSION = "row_commit_version";
+
+  // Metadata keys for row tracking metadata fields
+  private static final String BASE_ROW_ID_METADATA_COL_ATTR_KEY = "__base_row_id_metadata_col";
+  private static final String DEFAULT_ROW_COMMIT_VERSION_METADATA_COL_ATTR_KEY =
+      "__default_row_version_metadata_col";
+  private static final String ROW_ID_METADATA_COL_ATTR_KEY = "__row_id_metadata_col";
+  private static final String ROW_COMMIT_VERSION_METADATA_COL_ATTR_KEY =
+      "__row_commit_version_metadata_col";
+
+  private RowTrackingUtils() {}
+
+  /**
+   * Check if row tracking is enabled for reading.
+   *
+   * @param protocol the protocol to check
+   * @param metadata the metadata to check
+   * @return true if row tracking is enabled
+   * @throws IllegalStateException if row tracking is enabled in metadata but not supported by
+   *     protocol
+   */
+  public static boolean isEnabled(Protocol protocol, Metadata metadata) {
```

Comment:
From the PR description:
> The methods are spark specific logic(the metadata column names are not in delta protocol) so put it in connector folder in preparation integrating spark's parquet reader prototyped in https://github.com/delta-io/delta/pull/5484

That's not entirely true, a lot of this is part of the Delta spec (https://github.com/delta-io/delta/blob/8e72700b497caa82ddd14daf2b4f6258cd7e25e6/PROTOCOL.md#row-tracking) and should be in core kernel to make it reusable.
This `isEnabled` is a good example:  it doesn't depend on spark at all and would be useful in general in kernel, for example in https://github.com/delta-io/delta/blob/8e72700b497caa82ddd14daf2b4f6258cd7e25e6/kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/RowTracking.java#L248